client/state/themes/reducer.js

```diff
@@ -162,6 +162,26 @@ export function themeRequests( state = {}, action ) {
 }
 
 /**
+ * Returns the updated site theme requests error state after an action has been
+ * dispatched. The state reflects a mapping of site ID, theme ID pairing to a
+ * object describing request error. If there is no error null is storred.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const themeRequestsError = createReducer( {}, {
+	[ THEME_REQUEST_FAILURE ]: ( state, { siteId, themeId, error } ) => ( {
+		...state,
+		[ siteId ]: Object.assign( {}, state[ siteId ], { [ themeId ]: { error } } )
```

Comment:
This will produce a `themeRequestsError ` state like this (stolen from your unit tests 😛 ):

```js
2916284: {
	vivaro: {
		error: 'Request error'
	}
 }
```

Let's drop the error layer since it bears no extra information:

```js
2916284: {
	vivaro: 'Request error'
 }
```

i.e. just omit the curly braces around error so it doesn't produce an object:

```js
[ siteId ]: Object.assign( {}, state[ siteId ], { [ themeId ]: error } )
```

which I think you can then write using destructuring as

```js
[ siteId ]: {
	...state[ siteId ],
	[ themeId ]: error
}

```